client/cgo/cgotest.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+/*
+int testVerifyVrf(unsigned char *pk, int pkSize,
+    unsigned char *m, int mSize,
+    unsigned char *index, int indexSize,
+    unsigned char *proof, int proofSize) {
+    return cgoVerifyVrf(pk, pkSize, m, mSize, index, indexSize, proof, proofSize);
+}
+
+int testVerifySignature(
+    unsigned char *pk, int pkSize,
+    unsigned char *m, int mSize,
+    unsigned char *sig, int sigSize) {
+    return cgoVerifySignature(pk, pkSize, m, mSize, sig, sigSize);
+}
+
+int testHashChainInvalidSize() {
```

Comment:
What's going on in this test?  It kind seems like `cgoVerifyHashChain` should be taking `prevHash` (which comes from the data for the new STR) and `root` (which is the signature the client has stored, the previous STR) and then hashes `root` and does the byte comparison.